File: app/src/main/java/com/beone/bestpractice/BaseApp.java
package com.beone.bestpractice;

import android.app.Application;
import android.os.StrictMode;

import leakcanary.LeakCanary;

public class BaseApp extends Application {
  public static BaseApp instance;

  @Override
  public void onCreate() {
    super.onCreate();
    instance = this;
  }


  public void enableStrictMode() {
    StrictMode.setThreadPolicy(new StrictMode.ThreadPolicy.Builder()
            .detectAll()
            .penaltyLog()
            .penaltyDeath()
            .build());
  }
}

File: app/src/main/java/com/beone/bestpractice/BaseApp.kt
package com.beone.bestpractice

import android.app.Application
import android.os.StrictMode
import com.beone.bestpractice.core.di.databaseModule
import com.beone.bestpractice.core.di.networkModule
import com.beone.bestpractice.core.di.repositoryModule
import com.beone.bestpractice.di.useCaseModule
import com.beone.bestpractice.di.viewModelFactory
import org.koin.android.ext.koin.androidContext
import org.koin.android.ext.koin.androidLogger
import org.koin.core.context.startKoin
import org.koin.core.logger.Level

class BaseApp : Application() {
    override fun onCreate() {
        super.onCreate()
        startKoin {
            androidLogger(Level.INFO)
            androidContext(this@BaseApp)
            modules(listOf(
                    databaseModule,
                    networkModule,
                    repositoryModule,
                    useCaseModule,
                    viewModelFactory
            )
            )
        }

    }
}